### ISSUE TITLE
Adding a new user does not show an alert when a password is not of the correct format.

### ISSUE THREAD

**Issue by NikolasHadjipanayi** [RUGSoftEng/2017-Winter-Summer-School-App#162]
It just gives the user added successfully message without the user being added.

**Comment by Daanra** [comment]
Yeah the entire alert system is a bit funky. I want to delete it entirely and want to make the API's return some JSON with a message that tells you want went wrong.
In that case we can make all internal API calls through AJAX and properly display the result. 

We would also remove the redirects in our API which shouldnt be there in the first place.
